src/Faithlife.WebRequests/Json/JsonResponseUtility.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Text.RegularExpressions;
+
+namespace Faithlife.WebRequests.Json
+{
+	internal static class JsonResponseUtility
+	{
+		/// <summary>
+		/// Returns true if the content type is a JSON content type.
+		/// </summary>
+		/// <param name="contentType">The Content-Type http header value.</param>
+		/// <returns>True if the content type is "application/json" or "application/schema+json".</returns>
+		public static bool IsJsonContentType(string contentType)
+		{
+			if (contentType == null)
+				throw new ArgumentNullException(nameof(contentType));
+
+			return s_jsonContentTypeRegex.IsMatch(contentType);
+		}
+
+		private static readonly Regex s_jsonContentTypeRegex = new Regex(@"^\s*application\/([^\s;]+\+)?json");
```

Comment:
```suggestion
		private static readonly Regex s_jsonContentTypeRegex = new Regex(@"^\s*application\/([^\s;]+\+)?json$");
```

Could be wise to anchor `json` to the end.

As a counter-example, [`geo+json-seq`](https://tools.ietf.org/html/rfc8142) does not represent a valid JSON document, as it is a concatenated sequence of JSON objects.

Reply:
Oh; I see that the old code used `StartsWith`. That probably wasn't ideal, but changing it might be a breaking change?

Reply:
Yes I was worried about making a breaking change. I could make a major version bump. Oh wait that would put it past version 0.. maybe since it's version 0 it's okay to make breaking changes so long as they're noted in the changes?

We don't want just `$` because you can put parameters after a semicolon. For example the actual content-type I'm dealing with is `application/vnd.oracle.resource+json; type=singular; charset=UTF-8`. I added that one to the test cases (as well as your non-json example) and went ahead and made the breaking change, slightly modified to `\s*($|;)`.

I noted the breaking change in the version history, although I'm not sure what to do about the version number .. or even if I should have made the breaking change. Thoughts?

Reply:
We could just go to version 1 (major version change).

Reply:
Per https://semver.org/#spec-item-4, versions 0.y.z are not expected to have a stable public API. Incrementing the minor version is fine.

Reply:
Hmm but I got two PR approvals I think before I made the breaking change -- maybe I'll just undo the breaking change in the name of shipping.  We can always make a breaking change later :). Although it might be worse to do later. Hm yeah I think I'll undo the breaking change for now and then maybe immediately open a PR with the breaking change for consideration. I like separating out that change anyway.

Reply:
`...json\s*(;|$)` perhaps.

Reply:
Haha continuing my stream of consciousness... it's not quite as simple as separating out two changes.. because this change makes the problem worse.

Reply:
Whoops, I should have refreshed; you already solved the semicolon.

Reply:
Okay. I kept the breaking change in this PR, but I made it a minor version change rather than a patch version change. Barring objection I'll merge soon (or feel free to click merge).